sct-commons/src/main/java/org/lfenergy/compas/sct/commons/SclService.java

```diff
@@ -197,12 +219,65 @@ public List<SclReportItem> updateDoInRef(SCL scd) {
     }
 
     @Override
-    public List<SclReportItem> manageMonitoringLns(SCL scd) {
-        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        return sclRootAdapter.streamIEDAdapters()
-                .filter(iedAdapter -> !iedAdapter.getName().contains(IED_TEST_NAME))
-                .map(IEDAdapter::manageMonitoringLns)
-                .flatMap(List::stream)
+    public List<SclReportItem> manageMonitoringLns(SCL scd, JAXBContext jaxbContext) {
+        errorHanlder.clear();
+        List<TExtRef> tExtRefs =  scd.getIED().stream()
+                .flatMap(tied -> new LdeviceService(new LnService()).getLdevices(tied).flatMap(tlDevice -> Stream.of(tlDevice.getLN0())))
+                .filter(TAnyLN::isSetInputs)
+                .flatMap(ln0 -> ln0.getInputs().getExtRef().stream())
+                .filter(tExtRef -> tExtRef.isSetServiceType() && tExtRef.isSetSrcCBName())
+                .toList();
+        iedService.getFilteredIeds(scd, ied -> !ied.getName().contains(IED_TEST_NAME))
+                        .forEach(tied ->
+                                ldeviceService.findLdevice(tied, tlDevice -> LDEVICE_LDSUIED.equals(tlDevice.getInst()))
+                                .ifPresent(tlDevice -> {
+                                    manageMonitoringLns(tExtRefs.stream().filter(tExtRef -> TServiceType.GOOSE.equals(tExtRef.getServiceType())).toList(), scd, tied, tlDevice, DO_GOCBREF, MonitoringLnClassEnum.LGOS, jaxbContext);
+                                    manageMonitoringLns(tExtRefs.stream().filter(tExtRef -> TServiceType.SMV.equals(tExtRef.getServiceType())).toList(), scd, tied, tlDevice, DO_SVCBREF, MonitoringLnClassEnum.LSVS, jaxbContext);
+                                }));
+        return errorHanlder;
+    }
+
+    private void manageMonitoringLns(List<TExtRef> tExtRefs, SCL scd, TIED tied, TLDevice tlDevice, String doName, MonitoringLnClassEnum monitoringLnClassEnum, JAXBContext jaxbContext) {
+        List<TLN> lnList = lnService.getFilteredLns(tlDevice, tln -> monitoringLnClassEnum.value().equals(tln.getLnClass().getFirst())).toList();
+        if (lnList.isEmpty())
+            errorHanlder.add(SclReportItem.warning(tied.getName()+"/"+tlDevice.getInst()+"/"+monitoringLnClassEnum.value(), "There is no LN %s present in LDevice".formatted(monitoringLnClassEnum.value())));
+        if (tExtRefs.isEmpty()) {
+            return;
+        }
+        DoLinkedToDaFilter doLinkedToDaFilter = new DoLinkedToDaFilter(doName, List.of(), DA_SETSRCREF, List.of());
+        lnList.forEach(tln -> dataTypeTemplateService.getFilteredDoLinkedToDa(scd.getDataTypeTemplates(), tln.getLnType(), doLinkedToDaFilter)
```

Comment:
We should avoid these complicated methods;  
For the ExtRefs, we can inspire from this method : `com.rte_france.rspace.rconf.service.ExtRefsService#existsDoInLn` in RCONF